### PR TITLE
Bugfix/delete settings meta

### PR DESCRIPTION
**Pull request by forslund**
## Description
Fixes the incorrect variable name reported by @cclauss 
This PR also adds a couple of simple unittests for these methods and does a bit of simplification of the api tests.

## How to test
Check that the unittests passes.

## Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

File: test/unittests/api/test_api.py [hunk 1]
```diff
@@ -331,6 +308,88 @@ def test_has_been_paired(self, mock_identity_load):
...
+        self.mock_config_get = patcher.start()
+        self.addCleanup(patcher.stop)
+        super().setUp()
+
+    def test_upload_meta(self, mock_request, mock_identity_get):
+        mock_request.return_value = create_response(200, {})
+        mock_identity = mock.MagicMock()
+        mock_identity.is_expired.return_value = False
+        mock_identity.uuid = '1234'
+        mock_identity_get.return_value = mock_identity
```
Comment (cclauss) [inline test/unittests/api/test_api.py]: 4 lines —> 1 line: __mock_identity_get.return_value = create_identity('1234')__

File: test/unittests/api/test_api.py [hunk 2]
```diff
@@ -331,6 +308,88 @@ def test_has_been_paired(self, mock_identity_load):
...
+        self.assertEquals(params['json'], settings_meta)
+        self.assertEquals(
+            url, 'https://api-test.mycroft.ai/v1/device/1234/skill')
+
+    def test_get_skill_settings(self, mock_request, mock_identity_get):
+        mock_request.return_value = create_response(200, {})
+        mock_identity = mock.MagicMock()
+        mock_identity.is_expired.return_value = False
+        mock_identity.uuid = '1234'
+        mock_identity_get.return_value = mock_identity
```
Comment (cclauss) [inline test/unittests/api/test_api.py]: 4 lines —> 1 line: __mock_identity_get.return_value = create_identity('1234')__

File: test/unittests/api/test_api.py [hunk 3]
```diff
@@ -331,6 +308,88 @@ def test_has_been_paired(self, mock_identity_load):
...
+        self.assertEquals(method, 'GET')
+        self.assertEquals(
+            url, 'https://api-test.mycroft.ai/v1/device/1234/skill')
+
+    def test_delete_meta(self, mock_request, mock_identity_get):
+        mock_request.return_value = create_response(200, {})
+        mock_identity = mock.MagicMock()
+        mock_identity.is_expired.return_value = False
+        mock_identity.uuid = '1234'
+        mock_identity_get.return_value = mock_identity
```
Comment (cclauss) [inline test/unittests/api/test_api.py]: 4 lines —> 1 line: __mock_identity_get.return_value = create_identity('1234')__

Comment (forslund) [reply test/unittests/api/test_api.py]: Good catch, totally missed replacing in the new tests.

**Pull request merged**